### DISCUSSION TITLE
HA 2022.4 e sensore di temperatura/umidità/pressione BOSCH BME280 collegato a Raspberry

### DISCUSSION THREAD

**Discussion by MarQua** [indomus/forum#1619] [HUB: Home Assistant]
Salve,
con la nuova release 2022.4 l'utilizzo del sensore in oggetto è stato deprecato.
Non mi è però chiaro le alterative, se ci sono, per poterlo utilizzare.
Utilizzo HA Core su Raspberry PI4 e sensore Bosh BME280 collegato via I2c.
Grazie

**Comment by indomus** [comment]
TI basta prendere un NodeMCU, collegarglielo e usare ESPHome per integrare con Home Assistant.

**Reply by MarQua** [reply]
Grazie mille!

**Reply by indomus** [reply]
prendi spunto da qui:
https://indomus.it/progetti/realizzare-un-bridge-gateway-bluetooth-wi-fi-autonomo-con-nodemcu-esp32-e-esphome/

(volendo puoi far convivere più cose, tipo il bridge bluetooth per intregrare altra roba + il tuo sensore)

**Comment by MarQua** [comment]
Ciao,
senza acquistare nulla sono riuscito a utilizzare il sensore Bosch BME280 collegatai via I2c al mio Raspberry.

Ho creato degli script Python salvati in /home/homeassistant/.homeassistant/scripts
Riporto quello relativo alla Temperatura che l'ho creato modificando quello che mi ha fornito il produttore del chip  (per umidità e pressione si ottengono similmente)
[bme280_for_HA_temp.zip](https://github.com/indomus/forum/files/8457319/bme280_for_HA_temp.zip)

Poi ho editato il file configuration.yaml aggiungendo

sensor:
  - platform: command_line
    name: temp_interna
    command: "python3 /home/homeassistant/.homeassistant/scripts/bme280_for_HA_temp.py"
    unit_of_measurement: "°C"

Ho così creato l'entità sensor.temp_interna

Spero possa essere di aiuto a qualcuno.
